### DISCUSSION TITLE
[Suggestion] Changes to steel storage cabinets

### DISCUSSION THREAD

**Discussion by Singularia** [FakeFishGames/Barotrauma#11109] [Ideas and suggestions]
Suggestion to add a new category (or add to the existing one that can be related) in the submarine editor to include all storage containers (such as cabinets, compartments, crate/ammo racks, fire extinguisher, weapon holders etc).

Currently all containers except the **"Medium Windowed Steel Cabinet/Diving Gear Cabinet"**, which has `category="Diving"`, have no defined category in the editor.

This in my opinion will make it easier to quickly find the right container (as they will all be in one, preferably separate, category) if you can't remember the name of the container to type into the search bar.

**Comment by Singularia** [comment]
Suggestion to add the ability to add a list of possible **"allowed links"** to all containers (currently says **"Allowed links: None"**) to give new players a clue that you can link storage to some machines.

Additionally, the allowed links options can be extended to allow all containers to be linked not only to the (med)fabricator/deconstructor, but also to each other.
for example: `allowedlinks="fabricator,deconstructor,medicalfabricator,locker,container"`

**Comment by Singularia** [comment]
Proposal to add a medium-sized version of a Steel Cabinet to the game. Currently there are 2 types of Steel Cabinets in the game (excluding the **"Secured Steel Cabinet"** aka "Safe") **"Large Steel Cabinet"** (_3-section wide_) and **"Medium (Windowed) Steel Cabinet"** (_section wide_).

My proposalls:

1. Rename the current **"Medium Steel Cabinet"** to **"Small Steel Cabinet"**
2. Reduce its capacity to 8-10 storage units.
3. Add a 2-section wide cabinet which will replace the current **"Medium Steel Cabinet"** and act as a medium sized storage solution in the "Steel Cabinet" line-up of storage units.

Example picture and XML:
![Baro Feature request3](https://user-images.githubusercontent.com/62324831/215830860-70c488b6-9499-4587-b27c-8ba936aa05de.png)

```xml
<?xml version="1.0" encoding="utf-8"?>
<Items>
  <Item name="" identifier="smallcabinet" category="Misc" allowedlinks="fabricator,deconstructor,medicalfabricator,locker,container" tags="largeitem,locker,container" linkable="true" pickdistance="150" waterproof="true" scale="0.5" isshootable="true">
    <Sprite texture="%ModDir%/Items/Containers/Cabinets_Steel.png" depth="0.835" sourcerect="484,0,151,362" origin="0.5,0.5" />
    <Body width="151" height="362" density="20" />
    <ItemContainer capacity="8" slotsperrow="4" canbeselected="true" hideitems="true" msg="ItemMsgInteractSelect">
      <GuiFrame relativesize="0.25,0.25" anchor="Center" style="ItemUI" />
      <Containable items="smallitem,mediumitem" />
    </ItemContainer>
  </Item>
  <Item name="" identifier="smallwindowedcabinet" category="Misc,Diving" allowedlinks="fabricator,deconstructor,medicalfabricator,locker,container" tags="largeitem,locker,container" linkable="true" pickdistance="150" waterproof="true" scale="0.5">
    <Sprite texture="%ModDir%/Items/Containers/Cabinets_Steel.png" depth="0.835" sourcerect="635,0,151,362" origin="0.5,0.5" />
    <Body width="151" height="362" density="20" />
    <ItemContainer capacity="8" slotsperrow="4" canbeselected="true" hideitems="true" msg="ItemMsgInteractSelect">
      <GuiFrame relativesize="0.25,0.25" anchor="Center" style="ItemUI" />
      <Containable items="smallitem,mediumitem" />
    </ItemContainer>
  </Item>
  <Item name="" identifier="mediumcabinet" category="Misc" allowedlinks="fabricator,deconstructor,medicalfabricator,locker,container" tags="largeitem,locker,container" linkable="true" pickdistance="150" waterproof="true" scale="0.5">
    <Sprite texture="%ModDir%/Items/Containers/Cabinets_Steel.png" depth="0.835" sourcerect="0,0,242,362" origin="0.5,0.5" />
    <Body width="242" height="362" density="40" />
    <ItemContainer capacity="15" canbeselected="true" hideitems="true" msg="ItemMsgInteractSelect">
      <GuiFrame relativesize="0.25,0.35" anchor="Center" style="ItemUI" />
      <Containable items="smallitem,mediumitem" />
    </ItemContainer>
  </Item>
  <Item name="" identifier="mediumwindowedcabinet" category="Misc,Diving" allowedlinks="fabricator,deconstructor,medicalfabricator,locker,container" tags="largeitem,locker,container" linkable="true" pickdistance="150" waterproof="true" scale="0.5">
    <Sprite texture="%ModDir%/Items/Containers/Cabinets_Steel.png" depth="0.835" sourcerect="242,0,242,362" origin="0.5,0.5" />
    <Body width="242" height="362" density="40" />
    <ItemContainer capacity="15" canbeselected="true" hideitems="true" msg="ItemMsgInteractSelect">
      <GuiFrame relativesize="0.25,0.35" anchor="Center" style="ItemUI" />
      <Containable items="smallitem,mediumitem" />
    </ItemContainer>
  </Item>
  <Item name="" identifier="largecabinet" category="Misc" allowedlinks="fabricator,deconstructor,medicalfabricator,locker,container" tags="largeitem,locker,container" linkable="true" pickdistance="150" waterproof="true" scale="0.5">
    <Sprite texture="%ModDir%/Items/Containers/Cabinets_Steel.png" depth="0.835" sourcerect="0,362,352,361" origin="0.5,0.5" />
    <ItemContainer capacity="30" slotsperrow="6" canbeselected="true" hideitems="true" msg="ItemMsgInteractSelect">
      <GuiFrame relativesize="0.25,0.45" anchor="Center" style="ItemUI" />
      <Containable items="smallitem,mediumitem" />
    </ItemContainer>
  </Item>
</Items>
```

Example of texture file for both windowed and standard versions and the complete bundle in form of a mod are attached.
[Cabinets Request.zip](https://github.com/Regalis11/Barotrauma/files/10549201/Cabinets.Request.zip)

![Cabinets_Steel](https://user-images.githubusercontent.com/62324831/215830349-845bb781-9c72-49d4-a3d3-d0395040271a.png)

**Comment by FoxHoleman** [comment]
A really good one! Need this in game
